modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/utils/openCustomDateModal.tsx

```diff
@@ -0,0 +1,170 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+import ClayDatePicker from '@clayui/date-picker';
+import ClayForm, {ClaySelectWithOption} from '@clayui/form';
+import ClayModal, {useModal} from '@clayui/modal';
+import {render} from '@liferay/frontend-js-react-web';
+import classNames from 'classnames';
+import {useId} from 'frontend-js-components-web';
+import {navigate} from 'frontend-js-web';
+import React, {useState} from 'react';
+
+const NOT_SELECTED_OPTION = {
+	label: `-- ${Liferay.Language.get('select-type')} --`,
+	value: 'not-selected',
+};
+
+function getDateRange(start: string | undefined, end: string | undefined) {
+	if (start && end) {
+		return `${start} - ${end}`;
+	}
+
+	return undefined;
+}
+
+function isValidDate(date: string) {
+	return !isNaN(new Date(date).getDate());
+}
+
+type Props = {
+	dateTypes: Array<{label: string; value: string}>;
+	filterUrl: string;
+	namespace: string;
+	selectedDateType: string | undefined;
+	selectedEndDate: string | undefined;
+	selectedStartDate: string | undefined;
+};
+
+export default function openCustomDateModal(props: Props) {
+	render(CustomDateModal, {...props}, document.createElement('div'));
+}
+
+function CustomDateModal({
+	dateTypes,
+	filterUrl,
+	namespace,
+	selectedDateType,
+	selectedEndDate,
+	selectedStartDate,
+}: Props) {
+	const {observer, onOpenChange, open} = useModal({
+		defaultOpen: true,
+		onClose: () => onOpenChange(false),
+	});
+
+	const dateTypeId = useId();
+	const rangeId = useId();
+
+	const [dateType, setDateType] = useState(selectedDateType);
+	const [endDate, setEndDate] = useState(selectedEndDate);
+	const [startDate, setStartDate] = useState(selectedStartDate);
+
+	const [hasError, setHasError] = useState(false);
+
+	const onSave = () => {
+		if (!isValidDate(startDate!) || !isValidDate(endDate!)) {
+			setHasError(true);
+
+			return;
+		}
+
+		const url = new URL(filterUrl);
+
+		url.searchParams.set(`${namespace}dateType`, dateType!);
+		url.searchParams.set(`${namespace}startDate`, startDate!);
+		url.searchParams.set(`${namespace}endDate`, endDate!);
+
+		navigate(url.toString());
+	};
+
+	if (!open) {
```

Comment:
Is this needed? I tested with this removed and it works fine.

Reply:
Um, probably no, will make sure of it and remove it 🥰